src/hyperlight_host/src/hypervisor/hyperlight_vm.rs

```diff
@@ -52,8 +91,124 @@ pub(crate) struct HyperlightVm {
 }
 
 impl HyperlightVm {
-    pub(crate) fn new(inner: Box<dyn Hypervisor>) -> Self {
-        Self { vm: inner }
+    /// Create a new HyperlightVm instance (will not run vm until calling `initialise`)
+    #[instrument(err(Debug), skip_all, parent = Span::current(), level = "Trace")]
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) fn new(
+        mem_regions: Vec<MemoryRegion>,
+        _pml4_addr: u64,
+        entrypoint: u64,
+        rsp: u64,
+        #[cfg_attr(not(any(kvm, mshv3)), allow(unused_variables))] config: &SandboxConfiguration,
```

Comment:
isn't this the same as `#[cfg_attr(target_os = "windows"), allow(unused_variables))]`

Reply:
yes good catch, will fix